input/system/azure/system.go

```diff
@@ -0,0 +1,302 @@
+						HighAvailabilityState: util.StringCustomTypePtrToString(v.Properties.HighAvailability.State),
+						ReplicationRole:       util.StringCustomTypePtrToString(v.Properties.ReplicationRole),
+					}
+					resourceID = *v.ID
+					break
+				}
+			}
+		}
+	}
+
+	// Server info: Cosmos DB (when server is not found within Flexible Server)
+	if resourceID == "" {
+		clientFactory, err := armcosmosforpostgresql.NewClientFactory(config.AzureSubscriptionID, credential, nil)
+		if err != nil {
+			server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error making a Cosmos DB client: %v", err)
+			logger.PrintError("Azure/System: Failed to make a Cosmos DB client: %v\n", err)
+			return
+		}
+		// Search a server from the list
+		pager := clientFactory.NewClustersClient().NewListPager(nil)
+		for pager.More() {
+			page, err := pager.NextPage(ctx)
+			if err != nil {
+				logger.PrintError("Azure/System: Failed to advance page of Cosmos DB cluster list: %v\n", err)
+				break
+			}
+			for _, v := range page.Value {
+				if v.ID != nil {
+					resourceIDParts := strings.Split(*v.ID, "/")
+					resourceGroup := resourceIDParts[4]
+					resourceType := resourceIDParts[7]
+					resourceName := resourceIDParts[8]
+
+					if config.AzureDbServerName == resourceName {
+						system.Info.Azure = &state.SystemInfoAzure{
+							Location:                 util.StringPtrToString(v.Location),
+							CreatedAt:                util.TimePtrToTime(v.SystemData.CreatedAt),
+							State:                    util.StringCustomTypePtrToString(v.Properties.State),
+							SubscriptionID:           config.AzureSubscriptionID,
+							ResourceGroup:            resourceGroup,
+							ResourceType:             resourceType,
+							ResourceName:             resourceName,
+							CitusVersion:             util.StringPtrToString(v.Properties.CitusVersion),
+							HighAvailabilityEnabled:  util.BoolPtrToBool(v.Properties.EnableHa),
+							CoordinatorStorageMB:     util.Int32PtrToInt(v.Properties.CoordinatorStorageQuotaInMb),
+							NodeStorageMB:            util.Int32PtrToInt(v.Properties.NodeStorageQuotaInMb),
+							CoordinatorVCores:        util.Int32PtrToInt(v.Properties.CoordinatorVCores),
+							NodeVCores:               util.Int32PtrToInt(v.Properties.NodeVCores),
+							CoordinatorServerEdition: util.StringPtrToString(v.Properties.CoordinatorServerEdition),
+							NodeServerEdition:        util.StringPtrToString(v.Properties.NodeServerEdition),
+							NodeCount:                util.Int32PtrToInt(v.Properties.NodeCount),
+						}
+						resourceID = *v.ID
+						break
+					}
+				}
+			}
+		}
+	}
+
+	if resourceID == "" {
+		server.SelfTest.MarkCollectionAspectWarning(state.CollectionAspectSystemStats, "unable to find the database server info")
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the Reader permission of the database is granted to the managed identity.")
+		return
+	}
+
+	// Server metrics
```

Comment:
Starting from here, it's getting the metrics. Available metrics are different between Cosmos DB and Flexible Server, but we can use the same logic here.